src/main.cpp

```diff
@@ -5038,7 +5038,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
         // Be shy and don't send version until we hear
         if (pfrom->fInbound)
-            pfrom->PushVersion();
+            connman.PushVersion(pfrom, GetAdjustedTime());
```

Comment:
@TheBlueMatt I'm not sure I see the reasoning for either. I'd expect us to use our best guess of time everywhere.